DisguiseUnityRenderStream/ObjectField.cs

```diff
@@ -0,0 +1,112 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Reflection.Emit;
+
+namespace Disguise.RenderStream
+{
+    /// <summary>
+    /// Contains a dynamically-generated collection of methods to set a property or field
+    /// of a given type.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <remarks>
+    /// This class can be used to avoid boxing operations for calling property or field
+    /// setters when the parameter type is defined statically.
+    /// </remarks>
+    static class DynamicSetterCache<T>
+    {
+        static readonly Dictionary<MemberInfo, Action<object, T>> k_Cache = new();
+
+        public static Action<object, T> GetSetter(MemberInfo member)
+        {
+            var newMethod = new DynamicMethod($"{typeof(T).FullName}.argument_set_{member.Name}",
+                returnType: null,
+                parameterTypes: new[]
+                {
+                    typeof(object),
+                    typeof(T)
+                });
+            var gen = newMethod.GetILGenerator();
+            
+            if (!k_Cache.TryGetValue(member, out var setter))
+            {
+                switch (member)
+                {
+                    case FieldInfo field:
+                    {
+                        gen.Emit(OpCodes.Ldarg_0);
+                        gen.Emit(OpCodes.Ldarg_1);
+                        gen.Emit(OpCodes.Stfld, field);
+                        gen.Emit(OpCodes.Ret);
+
+                        break;
+                    }
+                    case PropertyInfo property:
+                    {
+                        gen.Emit(OpCodes.Ldarg_0);
+                        gen.Emit(OpCodes.Ldarg_1);
+                        gen.Emit(OpCodes.Call, property.GetSetMethod());
+                        gen.Emit(OpCodes.Ret);
+
+                        break;
+                    }
+                    default:
+                        throw new ArgumentOutOfRangeException(nameof(member), member, null);
+                }
+
+                setter = (Action<object, T>)newMethod.CreateDelegate(typeof(Action<object, T>));
+                k_Cache.Add(member, setter);
+            }
+
+            return setter;
+        }
+    }
+
+    class ObjectField
+    {
+        public object target;
```

Comment:
Minor: While reworking classes, do we want to rename member variables to m_?  Not critical but I found myself o a few occasion searching for a variable that was in fact a member variable...

Reply:
I didn't knew there was an old convention :)  On one hand I think the new convention would be better, but on the other hand I totally understand that updating all the code everywhere would take a lot of time...  Since you are the one currently using that code the most I'd say go with what you think is the best.